catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/KMLTransformerImplTest.java

```diff
@@ -222,6 +249,71 @@ public void testPerformDefaultTransformationMultiLineStringLocation()
     assertThat(multiLineString.getGeometry().get(1), instanceOf(LineString.class));
   }
 
+  @Test
+  public void testPerformDefaultTransformationExtendedData()
+      throws CatalogTransformerException, DateTimeParseException {
+    MetacardImpl metacard = createMockMetacard();
+
+    metacard.setLocation(POINT_WKT);
+    metacard.setCreatedDate(METACARD_DATE);
+    metacard.setEffectiveDate(METACARD_DATE);
+    metacard.setExpirationDate(METACARD_DATE);
+    metacard.setModifiedDate(METACARD_DATE);
+    metacard.setTags(METACARD_TAGS);
+
+    final Set<AttributeDescriptor> attributeDescriptors =
+        metacard.getMetacardType().getAttributeDescriptors();
+
+    Placemark placemark = kmlTransformer.performDefaultTransformation(metacard);
+
+    final List<Data> dataList = placemark.getExtendedData().getData();
+
+    int dataCount = 0;
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      final String attributeName = attributeDescriptor.getName();
+      final Attribute attribute = metacard.getAttribute(attributeName);
+      if (attribute != null) {
+        dataCount++;
+      }
+    }
+
+    assertThat(dataList.size(), is(dataCount));
+
+    for (Data data : dataList) {
+      switch (data.getName()) {
+        case Core.ID:
+          assertThat(data.getValue(), is(METACARD_ID));
+          break;
+        case Core.TITLE:
+          assertThat(data.getValue(), is(METACARD_TITLE));
+          break;
+        case Core.LOCATION:
+          assertThat(data.getValue(), is(POINT_WKT));
+          break;
+        case Metacard.CONTENT_TYPE:
+          assertThat(data.getValue(), is(METACARD_TYPE));
+          break;
+        case Metacard.CONTENT_TYPE_VERSION:
+          assertThat(data.getValue(), is(METACARD_VERSION));
+          break;
+        case Core.METADATA:
+          assertThat(data.getValue(), is(METACARD_METADATA));
+          break;
+        case Core.METACARD_TAGS:
+          assertThat(
+              data.getValue(),
+              is(METACARD_TAGS.asList().get(0) + "," + METACARD_TAGS.asList().get(1)));
```

Comment:
@bdeining there's some trouble with using `METACARD_TAGS` as a List. `metacard.setTags` requires a Set, and just having `is(METACARD_TAGS)` would return `[item1,item2]` where the actual is `item1,item2`